reco_utils/recommender/tfidf/tfidf_utils.py

```diff
@@ -0,0 +1,400 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics.pairwise import linear_kernel
+from transformers import BertTokenizer
+import re, string, unicodedata
+import pandas as pd
+import numpy as np
+
+import nltk
+from nltk.stem.porter import PorterStemmer
+
+class TfidfRecommender:
+    """Term Frequency - Inverse Document Frequency (TF-IDF) Recommender
+
+    This class provides content-based recommendations using TF-IDF vectorization in combination with cosine similarity.
+    """
+
+    def __init__(
+        self,
+        tokenization_method='scibert',
+        id_col='cord_uid',
+        title_col='title',
+        k=5
```

Comment:
suggest not setting k in the instantiation of this object, but instead using it in the methods that are need it. this way you can perform cleaning and then get top 5, top 10, etc, without having to create a new object or manipulate internal attributes

Reply:
Agreed. This has been fixed in a recent commit.